operators/src/processing/rasterization/mod.rs

```diff
@@ -0,0 +1,1082 @@
+        query: RasterQueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> util::Result<BoxStream<'a, util::Result<RasterTile2D<Self::RasterType>>>> {
+        if let MultiPoint(points_processor) = &self.input {
+            let grid_resolution = match self.grid_size_mode {
+                GridSizeMode::Fixed => SpatialResolution {
+                    x: f64::max(self.spatial_resolution.x, query.spatial_resolution.x),
+                    y: f64::max(self.spatial_resolution.y, query.spatial_resolution.y),
+                },
+                GridSizeMode::Relative => SpatialResolution {
+                    x: f64::max(
+                        self.spatial_resolution.x * query.spatial_resolution.x,
+                        query.spatial_resolution.x,
+                    ),
+                    y: f64::max(
+                        self.spatial_resolution.y * query.spatial_resolution.y,
+                        query.spatial_resolution.y,
+                    ),
+                },
+            };
+
+            let tiling_strategy = self
+                .tiling_specification
+                .strategy(query.spatial_resolution.x, -query.spatial_resolution.y);
+            let tile_shape = tiling_strategy.tile_size_in_pixels;
+
+            let tiles = stream::iter(
+                tiling_strategy.tile_information_iterator(query.spatial_bounds),
+            )
+            .then(move |tile_info| async move {
+                let grid_spatial_bounds = tile_info
+                    .spatial_partition()
+                    .snap_to_grid(self.origin_coordinate, grid_resolution);
+
+                let grid_size_x =
+                    f64::ceil(grid_spatial_bounds.size_x() / grid_resolution.x) as usize;
+                let grid_size_y =
+                    f64::ceil(grid_spatial_bounds.size_y() / grid_resolution.y) as usize;
+
+                let vector_query = VectorQueryRectangle {
+                    spatial_bounds: grid_spatial_bounds.as_bbox(),
+                    time_interval: query.time_interval,
+                    spatial_resolution: grid_resolution,
+                };
+
+                let grid_geo_transform = GeoTransform::new(
+                    grid_spatial_bounds.upper_left(),
+                    grid_resolution.x,
+                    -grid_resolution.y,
+                );
+
+                let mut chunks = points_processor.query(vector_query, ctx).await?;
+
+                let mut grid_data = vec![0.; grid_size_x * grid_size_y];
+                while let Some(chunk) = chunks.next().await {
+                    for &coord in chunk?.coordinates() {
+                        if !grid_spatial_bounds.contains_coordinate(&coord) {
+                            continue;
+                        }
+                        let [y, x] = grid_geo_transform.coordinate_to_grid_idx_2d(coord).0;
+                        grid_data[x as usize + y as usize * grid_size_x] += 1.;
+                    }
+                }
```

Comment:
Looping over all coordinates also looks like blocking to me, so this should be inside spawn_blocking as well.

Reply:
https://github.com/geo-engine/geoengine/pull/683/commits/07e582ba672fe3453ad1d4670cabce9756c23a3e